packages/office-ui-fabric-react/src/components/DatePicker/DatePicker.styles.ts

```diff
@@ -1,17 +1,75 @@
 import { IDatePickerStyleProps, IDatePickerStyles } from './DatePicker.types';
+import { IStyle, normalize, getGlobalClassNames } from '../../Styling';
 
-export const getStyles = (props: IDatePickerStyleProps): IDatePickerStyles => {
-  const { className } = props;
+const GlobalClassNames = {
+  root: 'ms-DatePicker',
+  textField: 'ms-TextField',
+  eventWithLabel: 'ms-DatePicker-event--with-label',
+  eventWithoutLabel: 'ms-DatePicker-event--without-label'
+};
+
+export const styles = (props: IDatePickerStyleProps): IDatePickerStyles => {
+  const { className, theme } = props;
+  const { palette, fonts } = theme;
+  const classNames = getGlobalClassNames(GlobalClassNames, theme);
+
+  const DatePickerEvent: IStyle = {
+    color: palette.neutralSecondary,
+    fontSize: fonts.medium.fontSize,
+    lineHeight: '18px',
+    pointerEvents: 'none',
+    position: 'absolute',
+    right: '9px'
+  };
 
   return {
-    root: [
-      'ms-DatePicker',
+    root: [classNames.root, normalize, {}, className],
+    textField: [
+      classNames.textField,
       {
-        // Insert css properties
+        position: 'relative',
+        selectors: {
+          '& input[readonly]': {
+            cursor: 'pointer'
+          },
+          input: {
+            selectors: {
+              '::-ms-clear': {
+                display: 'none'
+              }
+            }
+          }
+        }
+      },
+      className
+    ],
+    eventWithLabel: [
+      classNames.eventWithLabel,
+      DatePickerEvent,
+      {
+        bottom: '5px',
+        selectors: {
+          ':not(.msDatePickerDisabled)': {
+            pointerEvents: 'initial',
+            cursor: 'pointer'
+          }
+        }
+      },
+      className
```

Comment:
Shouldn't className only be on root? I don't see it mixed in with other class names in old code

Reply:
Yes, the standard className prop is a custom class name just for the root, not all elements.